modules/standard/Socket.md

```diff
@@ -0,0 +1,235 @@
+# Socket Module API
+
+To those familiar with the Unix socket API, the method names will feel familiar, though their usage will be somewhat simpler than the raw Unix socket API. The Module will comprise of four records which will abstract lower order functions to higher and more intutive methods for user:
+- ipAddr : Address information storage structure.
+- tcpConn : A TCP Socket with established connection
+- tcpListener : A TCP Socket listening for connections
+- udpSocket : A connection less UDP Socket
+
+## Address Information
+
+`ipAddr` serves the purpose of storing address information in a single record irrespective of the protocol used which varies the buffer required to store address.
+
+To initialize an `ipAddr` there are several available overloads and defaults.
+
+**Method**
+```python
+proc ipAddr.init(host:string = "localhost", port:uint(16) = 8000, family:IPFamily = IPFamily.IPv4)
+```
+**Parameters**
+```
+host: string - address string in dot or color notation depending on family. "127.0.0.1" or "::1"
+port: uint(16) - port number for socket
+family: IPFamily - Internet Protocol version which can be IPFamily.IPv4 or IPFamily.IPv6
+```
+
+There are two more initialzers which can take in standard IP Addresses like loopback and broadcast as `host` which are defined as constants inside the `Sys` module.
+
+**Method**
+```python
+proc ipAddr.init(host:sys_in_addr_t, port: uint(16) = 8000)
+proc ipAddr.init(host:sys_in6_addr_t, port: uint(16) = 8000)
```

Comment:
I'm a little iffy on C types being the only way to infer the family value.  Would it be reasonable to expose wrapper Chapel types and use those instead?

Reply:
- For IPV4 i think it will be possible as `in_addr_t` as I can see is
  ```C
  typedef uint32_t in_addr_t;
  ```
- For IPV6 i think that will not be a breeze or even possible not sure
  ```C
  #if !__USE_KERNEL_IPV6_DEFS
  /* IPv6 address */
  struct in6_addr
    {
      union
        {
     	  uint8_t	__u6_addr8[16];
	  uint16_t __u6_addr16[8];
	  uint32_t __u6_addr32[4];
        } __in6_u;
  #define s6_addr			__in6_u.__u6_addr8
  #ifdef __USE_MISC
  # define s6_addr16		__in6_u.__u6_addr16
  # define s6_addr32		__in6_u.__u6_addr32
  #endif
    };
  #endif /* !__USE_KERNEL_IPV6_DEFS */
  ```

Reply:
If the type is available as an extern type in Chapel code, it should be straight-forward to have a type variable that points to the extern type, right?  Maybe something like this?
```chapel
extern type sys_in6_addr_t;
type ipv6Addr = sys_in6_addr_t;
```

Reply:
oh okay, that works but that won't be wrapper 🤔 though right?

Reply:
Sorry, that's what I meant by wrapper.  Not a wrapper function or a type that holds another type, more like an alias

Reply:
Yeah an `alias` will be great I will make the change for same.